src/common/snippets/src/op/subgraph.cpp

```diff
@@ -650,6 +628,13 @@ void Subgraph::control_flow_transformations(lowered::LinearIR& linear_ir,
     INTERNAL_OP_SCOPE(Subgraph);
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::op::control_flow_transformations")
 
+    // Domain optimization must be the first pass, because all other transformations may depend on PortDescriptor shapes
+    size_t loop_depth = 1;
+    lowered::pass::OptimizeDomain(loop_depth).run(linear_ir);
+    // todo: for discussion on review: we can move control_flow_transformations to LinearIR
```

Comment:
For me it looks logically. Moreover, we discussed offline a little bit. That ngraph shape infer and transformations are in Subgraph. Now shape infer on Linear IR in there, so let's move transformations on Linear IR to there too.

Just two concerns. Firstly, we should set generator to new LinearIR::method, since we need `vector_size`. Secondly and probably, we need to move `m_buffer_scratchpad` to LinearIR and add `LinearIR::get_buffer_scratchpad_size()` method

Reply:
Ok. Thanks for the tips. 
I'll remove the todo, but let's move the control flow pipeline in the next (final) PR [18563](https://github.com/openvinotoolkit/openvino/pull/18563).